packages/in_app_purchase/lib/src/billing_client_wrappers/purchase_wrapper.dart

```diff
@@ -0,0 +1,132 @@
+        typedOther.packageName == packageName &&
+        typedOther.purchaseTime == purchaseTime &&
+        typedOther.purchaseToken == purchaseToken &&
+        typedOther.signature == signature &&
+        typedOther.sku == sku &&
+        typedOther.isAutoRenewing == isAutoRenewing &&
+        typedOther.originalJson == originalJson;
+  }
+
+  @override
+  int get hashCode => hashValues(orderId, packageName, purchaseTime,
+      purchaseToken, signature, sku, isAutoRenewing, originalJson);
+
+  /// The unique ID for this purchase. Corresponds to the Google Payments order
+  /// ID.
+  final String orderId;
+
+  /// The package name the purchase was made from.
+  final String packageName;
+
+  /// When the purchase was made, as an epoch timestamp.
+  final int purchaseTime;
+
+  /// A unique ID for a given [SkuDetailsWrapper], user, and purchase.
+  final String purchaseToken;
+
+  /// Signature of purchase data, signed with the developer's private key. Uses
+  /// RSASSA-PKCS1-v1_5.
+  final String signature;
+
+  /// The product ID of this purchase.
+  final String sku;
+
+  /// True for subscriptions that renew automatically. Does not apply to
+  /// [SkuType.inapp] products.
+  ///
+  /// For [SkuType.subs] this means that the subscription is canceled when it is
+  /// false.
+  final bool isAutoRenewing;
+
+  /// Details about this purchase, in JSON.
+  ///
+  /// This can be used verify a purchase. See ["Verify a purchase on a
+  /// device"](https://developer.android.com/google/play/billing/billing_library_overview#Verify-purchase-device).
+  /// Note though that verifying a purchase locally is inherently insecure (see
+  /// the article for more details).
+  final String originalJson;
+}
+
+/// A data struct representing the result of a transaction.
+///
+/// Contains a potentially empty list of [PurchaseWrapper]s and a
+/// [BillingResponse] to signify the overall state of the transaction.
+///
+/// Wraps [`com.android.billingclient.api.Purchase.PurchasesResult`](https://developer.android.com/reference/com/android/billingclient/api/Purchase.PurchasesResult).
+@JsonSerializable()
+@BillingResponseConverter()
+class PurchasesResultWrapper {
+  PurchasesResultWrapper(
+      {@required BillingResponse this.responseCode,
+      @required List<PurchaseWrapper> this.purchasesList});
+
+  factory PurchasesResultWrapper.fromJson(Map map) =>
+      _$PurchasesResultWrapperFromJson(map);
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(other, this)) return true;
+    if (other.runtimeType != runtimeType) return false;
+    final PurchasesResultWrapper typedOther = other;
+    return typedOther.responseCode == responseCode &&
+        typedOther.purchasesList == purchasesList;
+  }
+
+  @override
+  int get hashCode => hashValues(responseCode, purchasesList);
+
+  /// The status of the operation.
+  ///
+  /// Unfortunately this can represent both the status of the "query purchase
```

Comment:
(Personal opinion)Unfortunately seems too casual for dart doc? I don't know if I can find any document to support my point.